### PR TITLE
[flutter_tools] fix test asset loading

### PR DESCRIPTION
**Pull request by jonahwilliams**
Ensure that flutter_tester has the asset directory so that it can load test assets, fix automated test that was indented wrong.

Part of https://github.com/flutter/flutter/issues/103724

### CODE REVIEW

**Review by dnfield — APPROVED** [review]

**Comment by jonahwilliams** [comment]
Actually this will impact g3, holding off for now

**Comment by jonahwilliams** [comment]
Updated this change so that we can have the google3 tester pass in the correct asset path and fix test asset loading

**Comment by jonahwilliams** [comment]
Now that https://github.com/flutter/engine/pull/33323 has rolled in this should be safe to reland.

File: packages/flutter_tools/lib/src/test/flutter_tester_device.dart [hunk 1]
```diff
@@ -128,7 +127,10 @@ class FlutterTesterTestDevice extends TestDevice {
       '--enable-dart-profiling',
       '--non-interactive',
       '--use-test-fonts',
+      '--disable-asset-fonts',
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/test/flutter_tester_device.dart:130]: This is the new flag that prevents the tester from registering fonts bundled in the asset directory. It allows us to load other assets as normal while keeping the same font behavior

**Review by dnfield — APPROVED** [review]

**Pull request merged**